src/Microsoft.Diagnostics.Runtime/src/Common/ClrObject.cs

```diff
@@ -284,7 +290,7 @@ private ulong GetFieldAddress(string fieldName, ClrElementType element, out ClrT
             ClrType type = GetTypeOrThrow();
 
             if (IsNull)
-                throw new NullReferenceException();
+                throw new InvalidOperationException();
```

Comment:
This should also have an "Object is null" or similar exception message since we are throwing a more generic exception.  Maybe introduce a constant string for the class which is used in all places we would have thrown a NullReferenceException?

Reply:
I think I want to clean these up myself, there's more than just this one.